CHANGELOG.md

```diff
@@ -5,6 +5,12 @@ All notable changes to this crate are documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this crate adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [3.6.1]
+
+### Fixed
+
+- Deriving `Decode` will not trigger `redundant_closure_call` clippy warnings anymore
```

Comment:
And the unneeded return warning?

Reply:
Yes, no warnings should appear now. (Or at least they don't appear now when I tested it running `cargo clippy` in `polkadot` :P)